src/middlewares/auth.js

```diff
@@ -0,0 +1,119 @@
+import moment from 'moment';
+import jwt from 'jsonwebtoken';
+import ResponseGenerator from '../utils/response.util';
+import index from '../config/index';
+
+const response = new ResponseGenerator();
+/**
+   * @description - Authenticate middleware
+   */
+class Authentication {
+  /**
+   * @description - use for decoding token
+   * @param {Object} req
+   * @param {Object} res
+   * @param {Function} next
+   *
+   * @returns {Object} Object
+   */
+  static async authenticate(req, res, next) {
+    const payload = await Authentication.consumeToken(req);
```

Comment:
This should be wrapped in a `try` and `catch` so as to catch any unexpected error that may occur while using the `jwt` service

Reply:
OK. Thanks alot. Will do that.